File: backend/src/main/java/org/akashihi/mdg/api/v1/MdgException.java
package org.akashihi.mdg.api.v1;

import lombok.Data;
import lombok.EqualsAndHashCode;

@EqualsAndHashCode(callSuper = true)
@Data
public class MdgException extends RuntimeException {
    private final String code;
    public MdgException(String code) {
        super();
        this.code = code;
    }
    public MdgException(String code, Throwable cause) {
        super(cause);
        this.code = code;
    }
}

File: backend/src/main/kotlin/org/akashihi/mdg/api/v1/MdgException.kt
package org.akashihi.mdg.api.v1

class MdgException : RuntimeException {
    val code: String

    constructor(code: String) : super() {
        this.code = code
    }

    constructor(code: String, cause: Throwable?) : super(cause) {
        this.code = code
    }
}